osde2e/route_monitor_operator_tests.go

```diff
@@ -0,0 +1,250 @@
+
+		By("Checking the pod state")
+		var phase kubev1.PodPhase
+		err = wait.PollUntilContextTimeout(ctx, 15*time.Second, 2*time.Minute, false, func(ctx context.Context) (bool, error) {
+			err := k8s.Get(ctx, pod.Name, pod.Namespace, pod)
+			Expect(err).ShouldNot(HaveOccurred(), "Couldn't get the testing pod")
+			if pod != nil {
+				phase = pod.Status.Phase
+				if phase == kubev1.PodRunning || phase == kubev1.PodFailed {
+					return true, nil
+				}
+			}
+			GinkgoLogr.Info(fmt.Sprintf("Waiting for Pod '%s/%s' to be %s, currently %s...", pod.Namespace, pod.Name, kubev1.PodRunning, phase))
+			return false, nil
+		})
+		Expect(phase).To(Equal(kubev1.PodRunning), fmt.Sprintf("pod %s in ns %s is not running, current pod state is %v", routeMonitorName, namespace, phase))
+
+		By("Creating the service")
+		svc = &corev1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      routeMonitorName,
+				Namespace: namespace,
+				Labels:    map[string]string{routeMonitorName: routeMonitorName},
+			},
+			Spec: corev1.ServiceSpec{
+				Ports: []corev1.ServicePort{
+					{
+						Port:       8080,
+						Protocol:   corev1.ProtocolTCP,
+						TargetPort: intstr.FromInt(80),
+						Name:       "web",
+					},
+				},
+				Selector: map[string]string{routeMonitorName: routeMonitorName},
+			},
+		}
+
+		err = k8s.Create(ctx, svc)
+		Expect(err).ShouldNot(HaveOccurred(), "Unable to create service %s/%s", svc.Namespace, svc.Name)
+
+		By("Checking the route")
+		appRoute = &routev1.Route{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+				Name:      routeMonitorName,
+			},
+			Spec: routev1.RouteSpec{
+				To: routev1.RouteTargetReference{
+					Name: routeMonitorName,
+				},
+				TLS: &routev1.TLSConfig{Termination: "edge"},
+			},
+			Status: routev1.RouteStatus{},
+		}
+
+		By("Creating the route")
+		err = k8s.Create(ctx, appRoute)
+		Expect(err).ShouldNot(HaveOccurred(), "Unable to create route %s/%s", appRoute.Namespace, appRoute.Name)
+
+		By("Creating a sample RouteMonitor to monitor the service")
+		rmo = &routemonitorv1alpha1.RouteMonitor{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: namespace,
+				Name:      routeMonitorName,
+			},
+			Spec: routemonitorv1alpha1.RouteMonitorSpec{
+				Slo: routemonitorv1alpha1.SloSpec{
+					TargetAvailabilityPercent: "99.95",
+				},
+				Route: routemonitorv1alpha1.RouteMonitorRouteSpec{
+					Namespace: namespace,
+					Name:      routeMonitorName,
+				},
+			},
+		}
+
+		err = k8s.Create(ctx, rmo)
+		Expect(err).ShouldNot(HaveOccurred(), "Unable to create RouteMonitor %s/%s", rmo.Namespace, rmo.Name)
+
+		By("Checking the dependent resources are created")
+		_, err = serviceMonitorsClient.Get(ctx, consoleName, metav1.GetOptions{})
+		Expect(err).ShouldNot(HaveOccurred(), "Could not get console serviceMonitor")
+		_, err = prometheusRulesClient.Get(ctx, consoleName, metav1.GetOptions{})
+		Expect(err).ShouldNot(HaveOccurred(), "Could not get console prometheusRule")
```

Comment:
We don't want to check `consoleName` here, that was a different test. See here https://github.com/openshift/osde2e/blob/c74d64bbb6a0d5322f6057be4f5096cbc64fd805/pkg/e2e/operators/routemonitor.go#L117-L121

https://github.com/openshift/osde2e/blob/c74d64bbb6a0d5322f6057be4f5096cbc64fd805/pkg/e2e/operators/routemonitor.go#L67-L80